libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -38,57 +39,64 @@ internal class RouteRefreshController(
 
     suspend fun refresh(routes: List<NavigationRoute>): List<NavigationRoute> {
         return if (routes.isNotEmpty()) {
-            val routeToRefresh = routes.first()
-            when (val validationResult = validateRoute(routeToRefresh)) {
-                RouteValidationResult.Valid -> {
-                    val result = routes.toMutableList()
-                    result[0] = tryRefreshingUntilRouteChanges(routeToRefresh)
-                    result
-                }
-                is RouteValidationResult.Invalid -> waitForever(validationResult.reason)
+            require(
```

Comment:
Actually, I wanted to ask. How do you usually deal with unexpected situations? Throw exceptions or just log and ignore them? Is the behaviour different for user-provided data and server-provided data? Do you trust the sources? Only some of them? None?

Reply:
it depends on each case and can be discussed. If we can utilize an issue and keep working with existing data we can just log it (for instance if the SDK cannot refresh the route because of inconsistent data - we're good to log), but for instance, if an inconsistent state prevents working of ActiveGuidance or FreeDrive because the user provides the wrong data we should be good to throw an exception.

Reply:
It turned out that I can just get rid of the check and get desired behaviour because under the hood there are a mechanism which ignores refresh if nothing has changed. With empty/null legs, nothing changes so nothing happens. It feels so good having specific tests and generic implementation 😃